includes/subscriptions/class-wc-payments-subscriptions.php

```diff
@@ -94,11 +112,34 @@ public static function get_invoice_service() {
 	}
 
 	/**
-	 * Returns the the subscription service instance.
+	 * Returns the subscription service instance.
 	 *
 	 * @return WC_Payments_Subscription_Service
 	 */
 	public static function get_subscription_service() {
 		return self::$subscription_service;
 	}
+
+	/**
+	 * Returns the data migration manager instance.
+	 *
+	 * @return WC_Payments_Subscription_Data_Migration_Manager
+	 */
+	public static function get_data_migration_manager() {
+		if ( is_null( self::$data_migration_manager ) && did_action( 'init' ) ) {
+			self::$data_migration_manager = new WC_Payments_Subscription_Data_Migration_Manager();
+		}
+
+		return self::$data_migration_manager;
+	}
+
+	/**
+	 * Checks if the current request is to activate the WC Subscriptions plugin.
+	 *
+	 * @return bool
+	 */
+	private static function is_plugin_being_activated() {
+		$wc_subscriptions_plugin_slug = 'woocommerce-subscriptions/woocommerce-subscriptions.php';
+		return isset( $_GET['action'], $_GET['plugin'] ) && 'activate' === wc_clean( wp_unslash( $_GET['action'] ) ) && wc_clean( wp_unslash( $_GET['plugin'] ) ) === $wc_subscriptions_plugin_slug; //phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
Echoing a question I raised earlier as it also seems relevant here - https://github.com/Automattic/woocommerce-payments/issues/2963#issuecomment-925513762

> Do we need to consider other methods of activating/deactivating the subscriptions plugin?
> e.g.
>
>- WP CLI (wp plugin activate)
>- Jetpack jetpack.com/support/plugin-management
>- ManageWP, InfiniteWP, MainWP

Reply:
On second thought, this is probably covered here https://github.com/Automattic/woocommerce-payments/pull/2914/files#diff-c4d58ebbe28936148c761e80dee55e23bbc37bbac81ef082ecb37231f1712987R73

Reply:
> Echoing a question I raised earlier as it also seems relevant here - #2963 (comment)

yeah, that's a really good question.

> On second thought, this is probably covered here

Yeah, we aren't relying on catching the activation event in this case, we just check between two requests if the plugin was active and compare it to the current plugin status. 

Essentially if you deactivate the plugin via the WP admin UI, you'll get the notice (re https://github.com/Automattic/woocommerce-payments/issues/2963), but if you change the plugin status via one of the other options, the upgrade should still kick in, you just won't see the warning.

I'll make sure to cross-post this to issue #2963 to answer that question there too.